### PR TITLE
#418: Allow subtypes of Certificate and X509Certificate on LeshanServerBuilder

### PR DESCRIPTION
**Pull request by AlexITC**
Signed-off-by: Alexis Hernandez <alexis22229@gmail.com>

### CODE REVIEW

File: leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/LeshanServerBuilder.java [hunk 1]
```diff
@@ -226,15 +226,15 @@ public LeshanServerBuilder setPrivateKey(PrivateKey privateKey) {
      * For RPK the public key will be extract from the first X509 certificate of the certificate chain. If you only need
      * RPK support, use {@link LeshanServerBuilder#setPublicKey(PublicKey)} instead.
      */
-    public LeshanServerBuilder setCertificateChain(X509Certificate[] certificateChain) {
+    public <T extends X509Certificate> LeshanServerBuilder setCertificateChain(T[] certificateChain) {
```
Comment (sophokles73) [inline leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/LeshanServerBuilder.java]: pardon my ignorance but what is the difference between the two? What kind of certificates can I pass in now which I couldn't pass in before?

Comment (sbernard31) [reply leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/LeshanServerBuilder.java]: I thought that arrays were invariant like generics but [there are not](https://stackoverflow.com/questions/2745265/is-listdog-a-subclass-of-listanimal-why-arent-javas-generics-implicitly-p/46496077#46496077).
So good question @AlexITC  can you provide an example where this modification helps ?

Comment (AlexITC) [reply leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/LeshanServerBuilder.java]: oh, I forget to mention, I'm using Scala, Java arrays are handled as `Array<T>` collection, hence invariant.

**Comment by sophokles73** [comment]
LGTM

**Comment by sbernard31** [comment]
integrated in master (commit e54f0a368d2df10525bb75b9b902779c17917348)

**Pull request closed without merging**